src/topology/algebra/order/liminf_limsup.lean

```diff
@@ -185,3 +185,53 @@ end
 end conditionally_complete_linear_order
 
 end liminf_limsup
+
+section monotone
+
+variables {ι R S : Type*} {F : filter ι} [ne_bot F]
+  [complete_linear_order R] [topological_space R] [order_topology R]
+  [complete_linear_order S] [topological_space S] [order_topology S]
+
+/-- A continuous antitone function between complete linear ordered spaces sends a `filter.limsup`
+to the `filter.liminf` of the images. -/
+lemma antitone.map_limsup_of_continuous
+  (a : ι → R) {f : R → S} (f_decr : antitone f) (f_cont : continuous f ) :
+  f (F.limsup a) = F.liminf (f ∘ a) :=
```

Comment:
So, I think this (and the lemmas below) holds in greater generality, but not with the proof supplied here. In fact, if I'm not mistaken, I think this holds so long as `continuous_at f (F.limsup a)`. However, the proof given here wouldn't work because of the `congr_arg`. I'm not sure how hard it would be to do the generalization, so I am fine with leaving it as is, but if you can come up with a proof that isn't too hard, then go for it.

Reply:
You're probably right.

I thought a bit, and I don't think the lemmas included in this PR can be used for the proof of the stronger statement you suggest (or like you said, the `congr' in this proof is a wrong turn in the proof under the weaker hypotheses). The step by step argument currently pulls different supremums/infimums out from the function `f`, and therefore uses continuity of `f` at more than one point.

I would therefore like to leave possible generalizations for later, particularly since the next implication of portmanteau theorem depends on having continuous antitone liminf/limsup lemmas in at least some generality.

Reply:
The following proof does the more general statement:
```lean
/-- An antitone function between complete linear ordered spaces sends a `filter.Limsup`
to the `filter.liminf` of the image is it is continuous at the `Limsup`. -/
lemma antitone.map_Limsup_of_continuous {F : filter R} [ne_bot F]
  {f : R → S} (f_decr : antitone f) (f_cont : continuous_at f (F.Limsup)) :
  f (F.Limsup) = F.liminf f :=
begin
  apply le_antisymm,
  { have A : {a : R | ∀ᶠ (n : R) in F, n ≤ a}.nonempty, from ⟨⊤, by simp⟩,
    rw [Limsup, (f_decr.map_Inf_of_continuous_at' f_cont A)],
    apply le_of_forall_lt,
    assume c hc,
    simp only [liminf, Liminf, lt_Sup_iff, eventually_map, set.mem_set_of_eq, exists_prop,
      set.mem_image, exists_exists_and_eq_and] at hc ⊢,
    rcases hc with ⟨d, hd, h'd⟩,
    refine ⟨f d, _, h'd⟩,
    filter_upwards [hd] with x hx using f_decr hx },
  { rcases eq_or_lt_of_le (bot_le : ⊥ ≤ F.Limsup) with h|Limsup_ne_bot,
    { rw ← h,
      apply liminf_le_of_frequently_le,
      apply frequently_of_forall,
      assume x,
      exact f_decr bot_le },
    by_cases h' : ∃ c, c < F.Limsup ∧ set.Ioo c F.Limsup = ∅,
    { rcases h' with ⟨c, c_lt, hc⟩,
      have B : ∃ᶠ n in F, F.Limsup ≤ n,
      { apply (frequently_lt_of_lt_Limsup (by is_bounded_default) c_lt).mono,
        assume x hx,
        by_contra',
        have : (set.Ioo c F.Limsup).nonempty := ⟨x, ⟨hx, this⟩⟩,
        simpa [hc] },
      apply liminf_le_of_frequently_le,
      exact B.mono (λ x hx, f_decr hx) },
    by_contra' H,
    obtain ⟨l, l_lt, h'l⟩ : ∃ l < F.Limsup, set.Ioc l F.Limsup ⊆ {x : R | f x < F.liminf f},
      from exists_Ioc_subset_of_mem_nhds ((tendsto_order.1 f_cont.tendsto).2 _ H)
        ⟨⊥, Limsup_ne_bot⟩,
    obtain ⟨m, l_m, m_lt⟩  : (set.Ioo l F.Limsup).nonempty,
    { contrapose! h',
      refine ⟨l, l_lt, by rwa set.not_nonempty_iff_eq_empty at h'⟩ },
    have B : F.liminf f ≤ f m,
    { apply liminf_le_of_frequently_le,
      apply (frequently_lt_of_lt_Limsup (by is_bounded_default) m_lt).mono,
      assume x hx,
      exact f_decr hx.le },
    have I : f m < F.liminf f := h'l ⟨l_m, m_lt.le⟩,
    exact lt_irrefl _ (B.trans_lt I) }
end
```
I am not really satisfied by the proof, but at least it seems to work. (It needs 
```lean
lemma frequently_lt_of_lt_Limsup {f : filter α} [conditionally_complete_linear_order α] {a : α}
  (hf : f.is_cobounded (≤) . is_bounded_default) (h : a < f.Limsup) : ∃ᶠ n in f, a < n :=
begin
  contrapose! h,
  simp only [not_frequently, not_lt] at h,
  exact Limsup_le_of_le hf h,
end

lemma frequently_lt_of_Liminf_lt {f : filter α} [conditionally_complete_linear_order α] {a : α}
  (hf : f.is_cobounded (≥) . is_bounded_default) (h : f.Liminf < a) : ∃ᶠ n in f, n < a :=
@frequently_lt_of_lt_Limsup (order_dual α) f _ a hf h
```
to be inserted on Line 517 of  `src/order/liminf_limsup`).

Reply:
Great, thank you! These seem like the desired generality. Although I had to re-replace `[order_closed_topology]` by `[order_topology]` for this proof to work, but that is a smaller issue in my opinion.

I added your lemmas and the main result --- and 3 other versions of it for monotone/antitone map_Limsup/map_Liminf combinations. I removed the old lemmas, because they are (at least with the `[order_topology]` assumption) direct consequences of your result.